### PR TITLE
Add cleanup scripts and batch user invite

### PR DESCRIPTION
**Pull request by gigxz**
## Description

Add data cleanup scripts and batch user invite script. These are not exactly what was run, but they are close. May be helpful for future migrations.

## Type of change
- [ ] Bug fix
- [x] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [ ] I have performed a self-review of my code
- [ ] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [ ] My code follows the style guidelines of this project (rubocop)
- [ ] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: drivers/hmis/lib/hmis_util/batch_user_invite.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        agency = agency_by_name[agency_name]
+        unless agency
+          puts "Skipping #{email}, agency '#{agency_name}' does not exist."
+          next
+        end
+        first_name = name.split.first
+        last_name = name.split[1..].join(' ') if name.split.count > 1
+
+        puts "#{dry_run ? '' : 'Inviting user:'} FIRST:#{first_name}, LAST:#{last_name}, AGENCY:#{agency.name}, EMAIL:#{email}"
+        attributes = { first_name: first_name, last_name: last_name, email: email.downcase, agency_id: agency.id }
```
Comment (eanders) [inline drivers/hmis/lib/hmis_util/batch_user_invite.rb:33]: No need to adjust, but I wonder if in a future version we should collect up the skips and output and drop it in a CSV or echo it all as an array at the end.

File: drivers/hmis/lib/hmis_data_cleanup/util.rb [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    include ArelHelper
+
+    # Assign Household ID where missing
+    def self.assign_missing_household_ids!
+      scope = Hmis::Hud::Enrollment.hmis.where(household_id: nil)
+      Rails.logger.info "Assigning household id to #{scope.size} enrollments"
+      scope.each do |enrollment|
+        hh_id = Hmis::Hud::Base.generate_uuid
+        without_papertrail_or_timestamps do
+          enrollment.update_columns(household_id: hh_id) # skips callbacks
```
Comment (eanders) [inline drivers/hmis/lib/hmis_data_cleanup/util.rb:25]: If this gets to be a large number, and upsert is probably worth doing.  You could just `scope.find_in_batches` and then `enrollment.assign_attributes` and finally "import" the batch with a conflict key of `id`.

File: drivers/hmis/lib/hmis_data_cleanup/util.rb [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        without_papertrail_or_timestamps do
+          Hmis::Hud::Client.update(grouped_clients.keys, grouped_clients.values)
+          Rails.logger.info "Updated #{grouped_clients.size} clients"
+        end
+      end
+    end
+
+    # Change ProjectID of a project. Used in staging. Probably should never be run in Prod.
+    def self.change_project_id!(old_id, new_id)
+      project = ::Hmis::Hud::Project.hmis.find_by(project_id: old_id)
```
Comment (eanders) [inline drivers/hmis/lib/hmis_data_cleanup/util.rb:112]: I wonder if it's worth throwing a `return if Rails.env.production? && ! force` or something similar here.

**Review by eanders — APPROVED** [review]
These look good.  I left some comments, but they could totally be implemented only when we need this again (or not at all) they should have no impact on results, only performance.

File: drivers/hmis/lib/hmis_data_cleanup/util.rb (hunk 2 above)
Comment (gigxz) [reply drivers/hmis/lib/hmis_data_cleanup/util.rb:25]: Added comment

File: drivers/hmis/lib/hmis_data_cleanup/util.rb (hunk 3 above)
Comment (gigxz) [reply drivers/hmis/lib/hmis_data_cleanup/util.rb:112]: Added

**Pull request merged**